### PR TITLE
#682 Servlet support

### PR DESCRIPTION
**Pull request by t-izbassar**
Added servlet support classes.

### CODE REVIEW

**Comment by 0crat** [comment]
[@yegor256](http://www.0crat.com/u/yegor256) please, pay attention to this pull request

**Comment by 0crat** [comment]
Job [#762](https://github.com/yegor256/takes/issues/762) is now in scope, role is `REV`

**Comment by t-izbassar** [comment]
I'm working on rewriting this with PDD puzzlers.

**Comment by 0crat** [comment]
This pull request [#762](https://github.com/yegor256/takes/issues/762) is assigned to [@g4s8](http://www.0crat.com/u/g4s8). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and [§10](http://datum.zerocracy.com/pages/policy.html#10) and when you decide to accept the changes, inform [@yegor256](http://www.0crat.com/u/yegor256) right in this ticket.

File: src/main/java/org/takes/facets/servlet/SvResponseWrap.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+     * @param response ServletResponse to print to.
+     * @throws IOException If fails.
+     */
+    public void print(final HttpServletResponse response) throws IOException {
+        final Iterable<String> head = this.origin.head();
+        for (final String header : head) {
+            final String[] split = header.split(":");
+            final String name = split[0];
+            final String value = split[1];
+            response.addHeader(name, value);
```
Comment (g4s8) [inline src/main/java/org/takes/facets/servlet/SvResponseWrap.java]: @t-izbassar it's improtant to set a status-code to servlet response  (read it from first item in `head`) and parse cookie-headers from `takes.Response` and apply it via `HttpServletResponse.addCookie()`

File: src/main/java/org/takes/facets/servlet/SvResponseWrap.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    @SuppressWarnings("PMD.AssignmentInOperand")
+    private static String read(final InputStream body) throws IOException {
+        try (ByteArrayOutputStream data = new ByteArrayOutputStream()) {
+            final byte[] buffer = new byte[SvResponseWrap.BUFFER_LENGTH];
+            // @checkstyle InnerAssignmentCheck (1 line)
+            for (int len; (len = body.read(buffer)) != -1;) {
+                data.write(buffer, 0, len);
+            }
+            return data.toString(Charset.defaultCharset().name());
```
Comment (g4s8) [inline src/main/java/org/takes/facets/servlet/SvResponseWrap.java]: @t-izbassar I think it's not correct. Response can in binary format or text in another encoding - not always `Charset.defaultCharset()`. Let's copy response body directly to `ServletResponse.getOutputStream()`, don't forget to `flush()` it after all.

File: src/main/java/org/takes/facets/servlet/AbstractTakesServlet.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+/**
+ * AbstractTakesServlet.
+ *
+ * @author Izbassar Tolegen (t.izbassar@gmail.com)
+ * @version $Id.$
+ * @since 2.0
+ */
+public abstract class AbstractTakesServlet extends HttpServlet {
+
+    private static final long serialVersionUID = 1L;
```
Comment (g4s8) [inline src/main/java/org/takes/facets/servlet/AbstractTakesServlet.java]: @t-izbassar I don't think it's a good idea to set it as `1L`, let's use random 64bit number instead. Please see https://stackoverflow.com/q/285793

File: src/main/java/org/takes/facets/servlet/SvEmpty.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+/**
+ * Empty AbstractTakesServlet.
+ *
+ * @author Izbassar Tolegen (t.izbassar@gmail.com)
+ * @version $Id.$
+ * @since 2.0
+ */
+final class SvEmpty extends AbstractTakesServlet {
+
+    private static final long serialVersionUID = 1L;
```
Comment (g4s8) [inline src/main/java/org/takes/facets/servlet/SvEmpty.java]: @t-izbassar see previous comment about `serialVersionUID`

File: src/main/java/org/takes/facets/servlet/SvResponseWrap.java (hunk 2 above)
Comment (t-izbassar) [reply src/main/java/org/takes/facets/servlet/SvResponseWrap.java]: yes, I'm need to be cautious about that. thanks for review. Actually I'll planning to rewrite this code in more parts and create other tickets with PDD.

**Comment by 0crat** [comment]
Order was successfully finished: +15 points just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8), total is [+3625](https://www.0crat.com/u/g4s8)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://datum.zerocracy.com/pages/policy.html#28): +15 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256), total is [+8565](https://www.0crat.com/u/yegor256)

**Comment by 0crat** [comment]
The job [#762](https://github.com/yegor256/takes/issues/762) is now out of scope

**Pull request closed without merging**